misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt

```diff
@@ -0,0 +1,77 @@
+package misk.jobqueue
+
+import misk.hibernate.Gid
+import misk.hibernate.Session
+import misk.tokens.TokenGenerator
+import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedDeque
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.inject.Singleton
+
+/**
+ * A fake implementation of [TransactionalJobQueue] intended for testing.
+ *
+ * A [FakeTransactionalJobQueue] adds helper methods to inspect the jobs in the queue and to
+ * trigger processing.
+ *
+ * Example usage might look like this:
+ * ```
+ * callApiThatEnqueuesJob()
+ * assertJobIsCorrect(fakeJobQueue.peekJobs(MY_QUEUE).first())
+ *
+ * fakeJobQueue.handleJobs()
+ * assertJobSideEffects()
+ * ```
+ */
+@Singleton
+class FakeTransactionalJobQueue @Inject constructor(
```

Comment:
I think we should get in the practice of writing tests for our fakes, if they are public fakes that we expect devs to use. In an ideal world, the test would also verify the fake behaves like the real, but that's probably overkill for now.

Reply:
Yep, definitely. I'll add some tests